src/providers/WeaponProvider.ts

```diff
@@ -11,8 +11,8 @@ export class WeaponProvider {
   public getWeapons(): Weapon[] {
```

Comment:
You can add the filter logic to this function. Have the param be an object with all possible filters as optional fields.
i.e.
```
interface filterOptions = {
  type: ArcheType;
  element: Element,
  ... etc
}
```
If any of them are set, filter all weapons down by that field.